Sources/Core/AWSClientRuntime/Middlewares/Sha256TreeHashMiddleware.swift

```diff
@@ -110,10 +112,10 @@ extension Sha256TreeHashMiddleware: HttpInterceptor {
     public typealias InputType = OperationStackInput
     public typealias OutputType = OperationStackOutput
 
-    public func modifyBeforeTransmit(context: some MutableRequest<Self.InputType, Self.RequestType>) async throws {
+    public func modifyBeforeSigning(context: some MutableRequest<Self.InputType, Self.RequestType>) async throws {
```

Comment:
SHA256 middleware should occur prior to signing rather than adding headers aftwards